src/stream/src/cache/managed_lru.rs

```diff
@@ -13,63 +13,221 @@
+            self.kv_heap_size_dec(key.estimated_size() + value.estimated_size());
+        }
+    }
 
-    fn deref(&self) -> &Self::Target {
-        &self.inner
+    pub fn update_epoch(&mut self, epoch: u64) {
+        self.inner.update_epoch(epoch);
     }
-}
 
-impl<K, V, S, A: Clone + Allocator> DerefMut for ManagedLruCache<K, V, S, A> {
-    fn deref_mut(&mut self) -> &mut Self::Target {
-        &mut self.inner
+    pub fn current_epoch(&mut self) -> u64 {
+        self.inner.current_epoch()
+    }
+
+    /// An iterator visiting all values in most-recently used order. The iterator element type is
+    /// &V.
+    pub fn values(&self) -> impl Iterator<Item = &V> {
+        self.inner.iter().map(|(_k, v)| v)
+    }
+
+    pub fn put(&mut self, k: K, v: V) -> Option<V> {
+        let key_size = k.estimated_heap_size();
+        self.kv_heap_size_inc(key_size + v.estimated_heap_size());
+        let old_val = self.inner.put(k, v);
+        if let Some(old_val) = &old_val {
+            self.kv_heap_size_dec(key_size + old_val.estimated_heap_size());
+        }
+        old_val
+    }
+
+    pub fn get_mut(&mut self, k: &K) -> Option<MutGuard<'_, V>> {
+        let v = self.inner.get_mut(k);
+        v.map(|inner| {
+            MutGuard::new(
+                inner,
+                &mut self.kv_heap_size,
+                &mut self.last_reported_size_bytes,
+                self.memory_usage_metrics.clone(),
+            )
+        })
+    }
+
+    pub fn get_mut_unsafe(&mut self, k: &K) -> Option<UnsafeMutGuard<V>> {
+        let v = self.inner.get_mut(k);
+        v.map(|inner| {
+            UnsafeMutGuard::new(
+                inner,
+                &mut self.kv_heap_size,
+                &mut self.last_reported_size_bytes,
+                self.memory_usage_metrics.clone(),
+            )
+        })
+    }
+
+    pub fn get<Q>(&mut self, k: &Q) -> Option<&V>
+    where
+        KeyRef<K>: Borrow<Q>,
+        Q: Hash + Eq + ?Sized,
+    {
+        self.inner.get(k)
+    }
+
+    pub fn peek_mut(&mut self, k: &K) -> Option<MutGuard<'_, V>> {
+        let v = self.inner.peek_mut(k);
+        v.map(|inner| {
+            MutGuard::new(
+                inner,
+                &mut self.kv_heap_size,
+                &mut self.last_reported_size_bytes,
+                self.memory_usage_metrics.clone(),
+            )
+        })
+    }
+
+    pub fn push(&mut self, k: K, v: V) -> Option<(K, V)> {
+        self.kv_heap_size_inc(k.estimated_heap_size() + v.estimated_heap_size());
+
+        let old_kv = self.inner.push(k, v);
+
+        if let Some((old_key, old_val)) = &old_kv {
+            self.kv_heap_size_dec(old_key.estimated_heap_size() + old_val.estimated_heap_size());
+        }
+        old_kv
+    }
+
+    pub fn contains<Q>(&self, k: &Q) -> bool
+    where
+        KeyRef<K>: Borrow<Q>,
+        Q: Hash + Eq + ?Sized,
+    {
+        self.inner.contains(k)
+    }
+
+    pub fn len(&self) -> usize {
```

Comment:
`len` is more like length or capacity, this interface is the number of cached
```suggestion
    pub fn number_of_cached(&self) -> usize {
```

Reply:
Rust convention is to use `len` as the number of elements in a collection